src/crosschainMap/defaultLists/optimism.ts

```diff
@@ -0,0 +1,91 @@
+import { CrosschainMap } from '..'
+import { MappedToken, SupportedChains, Token } from '../../types'
+import fetch from 'node-fetch'
+
+async function getTokens() {
+  const URL =
+    'https://raw.githubusercontent.com/ethereum-optimism/ethereum-optimism.github.io/master/optimism.tokenlist.json'
+  const data = await fetch(URL)
+  const jsonData = await data.json()
+  return jsonData.tokens as MappedToken[]
+}
+
+const groupBy = (array: any[], key: string): { [key: string]: Token[] } => {
+  // Return the end result
+  return array.reduce((result, currentValue) => {
+    // If an array already present for key, push it to the array. Else create an array and push the object
+    ;(result[currentValue[key]] = result[currentValue[key]] || []).push(currentValue)
+    // Return the current iteration `result` value, this will be taken as next iteration `result` value and accumulate
+    return result
+  }, {}) // empty object is the initial value for result object
+}
+
+export const getOptimismTokens = async (crosschainMap: CrosschainMap) => {
+  console.log('OPTIMISM: get Optimism Tokens')
+  const L1L2Pairs = await getTokenPair(crosschainMap)
+  L1L2Pairs.forEach(({ tokenA, tokenB }) => crosschainMap.addPair(tokenA, tokenB))
+}
+
+async function getTokenPair(crosschainMap: CrosschainMap): Promise<
+  {
+    tokenA: Token
+    tokenB: Token
+  }[]
+> {
+  console.log('OPTIMISM: get Token Pairs')
```

Comment:
```suggestion
  console.log('OPTIMISM: get Token Pairs')
```
Do we need to console log this?

Reply:
I see so much data formatting you are doing here. See if Map or Set will help in reducing some looping?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
What would really help would be mapping the values.
I could use Map to get the entries and manipulate the values. After that I would have to reconstruct the Map/Map-like object.
I kind of agree, there's a lot of logic there and can be a bit difficult to parse but I think the benefit from using a Map is very small here.